### PR TITLE
`Development`: Refer to course enrollment instead of course registration when editing courses

### PR DESCRIPTION
**Pull request by pzdr7**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

The naming of "course registration" was not fully consistent. (Sometimes "registration", sometimes "enrollment").
image

### Description

- Adjusts translations to refer to enrollment
- Adjusts the form fields of the course update component to refer to enrollment instead of registration
- Removes code that manually transferred data from 'registration' fields of the form to 'enrollment' fields of the course
- Preserves the behavior of clearing the confirmation message if enrollment is not enabled.

### Steps for Testing

Prerequisites: 
- 1 Instructor with a course

1. Log in to Artemis
2. Navigate to Course Management > your course
3. Edit your course
4. Verify that you can edit the fields "Student Enrollment Enabled" and "Enrollment Confirmation Message".
5. Verify that this information is saved when you save the course.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

image

## Summary by CodeRabbit

- **New Features**
  - Renamed all instances of "registration" to "enrollment" in the course management interface for clarity and consistency.
  - Updated guided tour texts and course detail messages to reflect the terminology change from "registration" to "enrollment."

- **Refactor**
  - Improved logic handling for enrollment and course activation processes.

- **Documentation**
  - Updated German and English language files to align with new enrollment terminology.

### CODE REVIEW

File: src/main/webapp/app/course/manage/course-update.component.ts [hunk 1]
```diff
@@ -296,12 +296,8 @@ export class CourseUpdateComponent implements OnInit {
...
         }
 
-        // TODO: this has to be removed once the refactoring from course 'registration' to 'enrollment' is complete
-        course['enrollmentEnabled'] = course['registrationEnabled'];
-        delete course['registrationEnabled'];
-        if (course['enrollmentEnabled'] == true) {
-            course['enrollmentConfirmationMessage'] = course['registrationConfirmationMessage'];
-            delete course['registrationConfirmationMessage'];
+        if (!course['enrollmentEnabled']) {
+            course['enrollmentConfirmationMessage'] = undefined;
```
Comment (JohannesStoehr) [inline src/main/webapp/app/course/manage/course-update.component.ts:300]: Is there any reason to not use this?
```suggestion
        if (!course.enrollmentEnabled) {
            course.enrollmentConfirmationMessage = undefined;
```

**Review by JohannesStoehr — COMMENTED** [review]
Code looks generally good. Left one question:

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by Strohgelaender — DISMISSED** [review]
code lgtm

**Review by eceeeren — APPROVED** [review]
Tested on TS3, works fine
Code LGTM

**Review by rstief — DISMISSED** [review]
tested, could not find any occurrences of registration 👍

**Review by JohannesStoehr — APPROVED** [review]
Reapprove after merges

**Review by rstief — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**